index.ts

```diff
@@ -0,0 +1,217 @@
+            // 5. Check deviation
+            // **************************************************************************
+            const delta = beaconResponse.value.sub(apiValue).abs();
+            if (delta.eq(0)) {
+              console.log("Info: beacon is up-to-date. skipping update");
+              return;
+            }
+
+            const deviation = delta
+              .mul(100 * Number(reservedParameters._times)) // TODO: can _times be null or 0?
+              .div(apiValue);
+            console.log(
+              "Info: deviation %",
+              deviation.toNumber() / Number(reservedParameters._times)
+            );
+
+            // **************************************************************************
+            // 6. Update beacon if necessary (call makeRequest)
+            // **************************************************************************
+
+            // TODO: should we calculate the requestId hash or find a way to prevent sending
+            // the same request more that once? RrpBeaconServer.requestIdToTemplateId keeps
+            // track of the pending requests using a templateId
+
+            // TODO: 5% is hardcoded, should this be read from config?
+            const tolerance = 5;
+            if (deviation.lte(tolerance * Number(reservedParameters._times))) {
+              console.log(
+                "Info: delta between beacon and api value is within tolerance range. skipping update"
+              );
+              return;
+            }
+
+            /*
+             * 1. Airnode must first call setSponsorshipStatus(rrpBeaconServer.address, true) to
+             *    enable the beacon server to make requests to AirnodeRrp
+             * 2. Sponsor should then call setUpdatePermissionStatus(airnodeWallet.address, true)
+             *    to allow requester to update beacon
+             */
+            // console.log(
+            //   "🚀 ~ file: index.ts ~ line 161 ~ handler ~ await rrpBeaconServer.sponsorToUpdateRequesterToPermissionStatus()",
+            //   await rrpBeaconServer.sponsorToUpdateRequesterToPermissionStatus(
+            //     airnodeWallet.address,
+            //     airnodeWallet.address
+            //   )
+            // );
+
+            // TODO: who should be the sponsor?
+            // it's kinda weird to have to derive the sponsor wallet for the airnode wallet
+            const sponsorWalletAddress = deriveSponsorWallet(
+              ethers.utils.HDNode.fromMnemonic(
+                nodeSettings.airnodeWalletMnemonic
+              ),
+              airnodeWallet.address
+            ).address;
+
+            // TODO: why can't we send encoded parameters to be forwarded to AirnodeRrp?
+            // When using config.json.example we must pass a "from" parameter and the only
+            // way to get this request to work is if we add it a fixedParameter in the node
+            // config file
```

Comment:
The idea is that all parameters for a beacon is contained in the template, so the template has to be "full". Another thing to discuss at the call.

Reply:
👍🏻 I left the comment there as reminder that we need to discuss [this](https://api3workspace.slack.com/archives/C027Y2FAV0S/p1637335207072900) comment you made.